### PR TITLE
#135842529 search users endpoint

### PR DESCRIPTION
**Pull request by warenga**
#### What does this PR do?
Adds api/search/users endpoint
 #### Description of Task to be completed?
Returns the information of the searched user.
 #### How should this be manually tested?
 #### Any background context you want to provide?
It is an authenticated endpoint thus can only be accessed once you are authenticated. 
 #### What are the relevant pivotal tracker stories?
#135842529
 #### Screenshots (if appropriate)
postman screenshot
screen shot 2017-01-20 at 3 24 13 pm
 #### Questions:

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -298,4 +298,15 @@ exports.authToken = function(req, res, next) {
...
     }
+};
+
+/**
+ * Search current users by username
+ */ 
+exports.searchUsers = function(req, res) {
+  User.find({
+    name: new RegExp(req.query.username, 'i')
+  }, function(err, users) {
```
Comment (boswellgathu) [inline app/controllers/users.js:309]: Add error handling in case of any.

File: config/routes.js [hunk 2]
```diff
@@ -10,6 +10,9 @@ module.exports = function(app, passport, auth) {
     app.post('/api/auth/signup', users.register);
     app.post('/api/auth/signin', users.login);
 
+    // Search all current users
+    app.get('/api/search/users', users.authToken, users.searchUsers);
```
Comment (boswellgathu) [inline config/routes.js:14]: work with Boswell to make sure that when there is a token, it is actually authenticated

**Review by boswellgathu — CHANGES_REQUESTED** [review]
Make those changes then I will review

File: app/controllers/users.js [hunk 3]
```diff
@@ -307,6 +307,7 @@ exports.searchUsers = function(req, res) {
   User.find({
     name: new RegExp(req.query.username, 'i')
   }, function(err, users) {
+    if (err) return next(err);
```
Comment (warenga) [inline app/controllers/users.js:310]: @andela-Bgathu Thanks for the observation. I've added it.

**Comment by andela-kakpobome** [comment]
### Critical

@andela-wmaina why do we have a hashed password in the response?!

**Pull request closed without merging**